src/autopas/utils/RaplMeter.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file RaplMeter.cpp
+ * @author Vincent Fischer
+ * @date 09.11.2021
+ */
+
+#include "autopas/utils/RaplMeter.h"
+
+#include <linux/perf_event.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include <cmath>
+#include <cstring>
+#include <sstream>
+
+#include "autopas/utils/ExceptionHandler.h"
+#include "autopas/utils/logging/Logger.h"
+
+namespace autopas::utils {
+
+int RaplMeter::open_perf_event(int type, int config) {
+  struct perf_event_attr attr;
+  memset(&attr, 0, sizeof(attr));
+  attr.type = type;
+  attr.config = config;
+  if (config == 0) {
+    return -1;
+  }
+
+  int fd = syscall(__NR_perf_event_open, &attr, -1 /*PID*/, 0 /*CPU*/, -1 /*GROUP FD*/, 0 /*FLAGS*/);
+  if (fd < 0) {
+    if (errno == EACCES) {
+      throw ExceptionHandler::AutoPasException("Failed to open perf event: Permission denied");
+    }
+    throw ExceptionHandler::AutoPasException("Failed to open perf event");
+  }
+  // TODO: Error handling
+  return fd;
+}
+
+void RaplMeter::init() {
+  FILE *fff = fopen("/sys/bus/event_source/devices/power/type", "r");
+  if (fff == NULL) {
```

Comment:
```suggestion
  if (FILE *fff = fopen("/sys/bus/event_source/devices/power/type", "r"); not fff) {
```
1. The declaration can be scoped to the if (not 100% sure about syntax). This would make it a bit more easier to follow.
2. In most following blocks you compare `fff != NULL` but here  you compare `fff == NULL`. Probably better to always have the same logic for consistency.
3. No need to write the actual comparison. AFAIK `fff` should be a `nullptr` if `fopen` fails. `nullptr == 0 == false`